src/utils/arch/aarch64/cpu.h

```diff
@@ -35,4 +35,44 @@
 #define ucc_memory_cpu_store_fence()  ucc_aarch64_dmb(ishst)
 #define ucc_memory_cpu_load_fence()   ucc_aarch64_dmb(ishld)
 
+typedef struct ucc_aarch64_cpuid {
+    int       implementer;
+    int       architecture;
+    int       variant;
+    int       part;
+    int       revision;
+} ucc_aarch64_cpuid_t;
+
+static inline void ucc_aarch64_cpuid(ucc_aarch64_cpuid_t *cpuid);
+{
+    static ucc_aarch64_cpuid_t cached_cpuid;
+    static int initialized = 0;
+
+    if (!initialized) {
+        ucc_aarch64_cpuid_from_proc(&cached_cpuid);
```

Comment:
this functions is undefined, pls test on arm and ppc to make sure it works

Reply:
added and tested.